gama/gama.py

```diff
@@ -509,10 +509,17 @@ def fit(
                 self._operator_set._compile,
                 preprocessing_steps=self._fixed_pipeline_extension,
             )
-
             store_pipelines = (
                 self._evaluation_library._m is None or self._evaluation_library._m > 0
             )
+
+            # Add label information for classification to the scorer such that
+            # the cross validator does not encounter unseen labels in smaller
+            # data sets during pipeline evaluation.
+            for m in self._metrics:
+                if m.task_type == MetricType.CLASSIFICATION:
+                    m.scorer._kwargs.update({"labels": self._y})
```

Comment:
Since this is classification specific logic, it would be best to include this in `GamaClassifier` instead.

Reply:
I've moved the logic to `GamaClassifier.py` as discussed. One thing to note though, is that the formatting done to `y` by `format_x_y()` in `gama.py` would not be passed on to the `scorer`. But I'm sure that the formatting won't be that relevant to the `scorer`.

Reply:
The only case where this might matter is when `y` contains missing values, which is not a case we currently _officially_ support anyway.